### ISSUE TITLE
Highlights don't work correctly for groups any more

### ISSUE THREAD

**Issue by primaeval** [vonH/plugin.video.iplayerwww#6]
Highlights have gone a bit wierd. 
see "Series 9:1 The Magicians Apprentice" should have "Doctor Who" at the start.

**Comment by CaptainTK** [comment]
Actually, there is more broken. The Beeb have changed the website once again. This is getting kind of annoying for the Highlights section. Some new twist every other week.

**Comment by CaptainTK** [comment]
The current status of the Highlights page seems somewhat inconsistent to me:
* some episodes appear twice in the HTML code, but are displayed only once
* for some groups the group title is the series title, like Doctor Who or F1, for others it is not

My suggestion is to leave Highlights broken for a couple of days and then check again if these inconsistencies have been resolved by the Beeb. Until then, Highlights are broken.

**Comment by primaeval** [comment]
The simplest way would be just to pull the title off the end of the episode's href, remove the "-"s and capitalise it. Otherwise you would have to visit every link to find the proper title, I would have thought.

**Comment by CaptainTK** [comment]
It's actually even more complicated. Some shows appear twice in HTML, once as part of a group without a preview image and once with a preview image and an aired date and so on. If you just parse everything by href, you will end up with lots of duplicates (but not all of them). The stupid thing is: Some episodes only show up as part of a group without a preview image.

Getting to a clean solution will take some time and thinking. I would propose to leave Highlights broken for the time being or bring back minimal functionality (groups). I already have a patch for that part.

**Comment by CaptainTK** [comment]
This should be fixed in 1.1.2. However, currently there is no group on the Highlights page which breaks down into main title and episode title. I believe that these can be identified, but the code to do this is based on my memory. I currently have no way of testing it. Let's see what happens in the next days.

**Comment by primaeval** [comment]
Its nearly there. I spotted duplicates of Doctor Who 9.4. A set would be a quick way to get rid of duplicates. It would be nice if it was sorted too.

**Comment by CaptainTK** [comment]
I am not surprised. This episode actually shows up twice on the website. Once as part of Most Popular and once as part of Drama Picks.

Will sort that out next time round.

**Comment by primaeval** [comment]
I did a quick fix that seems to work for me. 
https://github.com/primaeval/plugin.video.iplayerwww/commit/bde10506e7434d8362fb7b1ec0d0ff0a25df8882#diff-e87f4dd869f897586689ccfd0fa643d5

**Comment by primaeval** [comment]
I found the duplicate in the first episodelist match. Here is my effort following your conventions. https://github.com/primaeval/plugin.video.iplayerwww/commit/74ea3a815d5a22146c9b2bb6cdeb1d2d9a972f62#diff-e87f4dd869f897586689ccfd0fa643d5

**Comment by primaeval** [comment]
Maybe this is the right link. https://github.com/vonH/plugin.video.iplayerwww/compare/master...primaeval:master

**Comment by primaeval** [comment]
I got a failed error this morning. Scream Street was missing an aired match.
Here is my fix:
https://github.com/primaeval/plugin.video.iplayerwww/commit/6abdb108cc62eef3f3e670142346672ad9e5abc7

**Comment by primaeval** [comment]
Highlights seem ok to me now. Do you want to close this?

**Comment by CaptainTK** [comment]
I'd like to wait for a series-catchup group to show up first to see if it works correctly for them as well.

My guess is that we will see one by the weekend, probably Doctor Who again.

**Comment by CaptainTK** [comment]
Just verified that it also works with series-catchup, so this issue can be closed. There will be minor tweaks for the display in the next commit, though.
